### ISSUE TITLE
Fail to compile rayon-demo

### ISSUE THREAD

**Issue by jeehoonkang** [nikomatsakis/rayon#398]
I failed to compile rayon-demo.  Here is the error message:

```
$ cd rayon-demo

$ cargo --version
cargo 0.21.0-nightly (ffab51954 2017-07-18)

$ rustc --version
rustc 1.20.0-nightly (afe145d22 2017-07-23)

$ cargo +nightly run --release -- nbody visualize
   Compiling regex v0.2.2
   Compiling glium v0.16.0
   Compiling glutin v0.7.4
   Compiling gleam v0.2.32
   Compiling core-graphics v0.3.2
error: type `color_space::__CGColorSpace` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/context.rs:91:48
   |
91 |                                                space.as_concrete_TypeRef(),
   |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type `color_space::__CGColorSpace` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/context.rs:86:26
   |
86 |             let result = CGBitmapContextCreate(ptr::null_mut(),
   |                          ^^^^^^^^^^^^^^^^^^^^^

error: type `color_space::__CGColorSpace` is private
   --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/context.rs:144:5
    |
144 | /     fn CGBitmapContextCreate(data: *mut c_void,
145 | |                              width: size_t,
146 | |                              height: size_t,
147 | |                              bitsPerComponent: size_t,
...   |
150 | |                              bitmapInfo: u32)
151 | |                              -> CGContextRef;
    | |_____________________________________________^

error: type `data_provider::__CGDataProvider` is private
   --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/font.rs:106:57
    |
106 |             let font_ref = CGFontCreateWithDataProvider(provider.as_concrete_TypeRef());
    |                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type `data_provider::__CGDataProvider` is private
   --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/font.rs:106:28
    |
106 |             let font_ref = CGFontCreateWithDataProvider(provider.as_concrete_TypeRef());
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type `data_provider::__CGDataProvider` is private
   --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/font.rs:137:5
    |
137 |     fn CGFontCreateWithDataProvider(provider: CGDataProviderRef) -> CGFontRef;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:19:5
   |
19 |     region: ffi::CGSRegionRef,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:25:35
   |
25 |             ffi::CGSRegionRelease(self.region)
   |                                   ^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:25:13
   |
25 |             ffi::CGSRegionRelease(self.region)
   |             ^^^^^^^^^^^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:34:30
   |
34 |             let mut region = ptr::null_mut();
   |                              ^^^^^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:35:53
   |
35 |             assert!(ffi::CGSNewRegionWithRect(rect, &mut region) == 0);
   |                                                     ^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:35:21
   |
35 |             assert!(ffi::CGSNewRegionWithRect(rect, &mut region) == 0);
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:37:25
   |
37 |                 region: region,
   |                         ^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:45:30
   |
45 |             let mut region = ptr::null_mut();
   |                              ^^^^^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:48:51
   |
48 |                                                   &mut region) == 0);
   |                                                   ^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:46:21
   |
46 |             assert!(ffi::CGSNewRegionWithRectList(rects.as_ptr(),
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:50:25
   |
50 |                 region: region,
   |                         ^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:84:45
   |
84 |                                             region.region) == 0)
   |                                             ^^^^^^^^^^^^^

error: type `private::ffi::CGSRegionObject` is private
  --> /Users/jeehoon/.cargo/registry/src/github.com-1ecc6299db9ec823/core-graphics-0.3.2/src/private.rs:81:21
   |
81 |             assert!(ffi::CGSSetSurfaceShape(self.context_id,
   |                     ^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 19 previous errors

error: Could not compile `core-graphics`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by cbarrick** [comment]
The first thing I notice is that it's trying to compile [core-graphics](https://github.com/servo/core-graphics-rs) v0.3.2, which is about a year old. The current version is v0.8.2.  (as an aside, the demos *also* depend on core-graphics v0.4.2).

The dependency chain bringing in v0.3.2 seems to be

```
rayon_demo
depends on glium 0.16
depends on glutin ^0.7.1 (optional)
depends on cocoa ^0.3.2
depends on core-graphics >= 0.2, < 0.4
```

Here's the current dependency graph
![rayon-demo](https://user-images.githubusercontent.com/271982/28636213-6f914328-720b-11e7-9f2f-452f9ae23042.png)

I tried updating the glium dependency to 0.17. As a result, cargo pulls in the newest core-graphics, and succeeds in building it. But the demo then fails because `glium::DisplayBuild` no longer exists.

**Comment by cuviper** [comment]
I'd be happy to merge dependency updates for rayon-demo if you can get it working.

**Comment by cuviper** [comment]
Should be fixed by the updates in #436.
